### DISCUSSION TITLE
How to extract from the first commit

### DISCUSSION THREAD

**Discussion by sfukumaru-clara** [scolladon/sfdx-git-delta#664] [Q&A] [answered]
Hi, I'm trying to extract all metadata from the Git repository by using this plugin. However, I can not do it even if specifying the first commit(the package.xml file only contains from the second commit). So, is there any way to do it? Or should I insert empty commit as the first commit?

**Comment by scolladon** [comment, marked as answer]
Hi @sfukumaru-clara 

If you could have your first commit containing only scaffolding info for your project it would be best.
This way the metadata would be stored in the repo in the second commit and you will not face this issue.
There is no way for the plugin to compare from the init of the repo as there is no way to diff the first commit with the previous step via the plugin.

**Reply by sfukumaru-clara** [reply]
Hi scolladon, 

Thanks. My conclusion is same. Although I have to prepare a workaround this time,I will commit metadata from the second commit for the next time.
